Sources/RichTextKit/Coordinator/RichTextCoordinator+ColorSubscriptions.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  RichTextCoordinator+ColorSubscriptions.swift
+//
+//
+//  Created by Dominik Bucher on 18.01.2024.
+//
+
+#if iOS || macOS || os(tvOS)
+import Foundation
+
+extension RichTextCoordinator {
+    
+    func setColor(_ color: ColorRepresentable, for coloredAttribute: RichTextColor) {
+        var applyRange: NSRange? = nil
+        if textView.hasSelectedRange {
+            applyRange = textView.selectedRange
+        }
+        switch coloredAttribute {
+        case .foreground:
+            setForegroundColor(color, at: applyRange)
+        case .background:
+            setBackgroundColor(color, at: applyRange)
+        case .strikethrough:
+            setStrikethroughColor(color, at: applyRange)
+        case .stroke:
+            setStrokeColor(color, at: applyRange)
+        case .underline:
+            setUnderlineColor(color, at: applyRange)
+        case .undefined:
+            break
+        }
+    }
+    
+    private func setStrikethroughColor(_ color: ColorRepresentable, at range: NSRange?) {
```

Comment:
If the `RichTextColor` defines its attribute (I think it already does), then we don't need all these functions.